kernel-open/nvidia/nv-caps.c

```diff
@@ -594,9 +594,9 @@ static nv_cap_t* nv_cap_alloc(nv_cap_t *parent_cap, const char *name)
         return NULL;
     }
 
-    strcpy(cap->path, parent_cap->path);
-    strcat(cap->path, "/");
-    strcat(cap->path, name);
+    strncpy(cap->path, parent_cap->path, len);
+    strncat(cap->path, "/", len);
+    strncat(cap->path, name, len);
```

Comment:
This code might be misunderstanding what the third argument of `strncat` is. If the intention is to have a line of code that can never write `len` bytes past the start of `cap->path`", then the correct code is `strcat(cap->path, name, len - strlen(cap->path) - 1);`. Almost certainly better to leave the code as it is...

Reply:
My bad, will fix this when I get home, will reinstate strcat, I will also replace strncpy by by strscpy as suggested here before. 

in the files that are not linux-specific, i can first do a #ifdef linux, and keep using strncpy if the linux isn't defined and otherwise use the safer strscpy

Reply:
You could use strlcat instead of strcat:
https://linux.die.net/man/3/strlcat